llama.cpp

```diff
@@ -11165,6 +11241,23 @@ static void llama_set_inputs(llama_context & lctx, const llama_batch & batch) {
         }
     }
 
+    if (lctx.freq_factors) {
+        // TODO: this might have to be hparams.n_rot instead of hparams.n_embd_head_k, but maybe it does not matter
+        const auto freq_dim = hparams.n_embd_head_k / 2;
+
+        GGML_ASSERT(lctx.freq_factors->ne[0] == freq_dim);
+        GGML_ASSERT(hparams.rope_long_factors.size()  == freq_dim);
+        GGML_ASSERT(hparams.rope_short_factors.size() == freq_dim);
+
+        // choose long/short freq factors based on the context size
+        const auto n_ctx = llama_n_ctx(&lctx);
```

Comment:
Would this work correctly with multiple sequences? Maybe something like `llama_n_ctx(&lctx) / llama_n_seq_max(&lctx)` would be correct in more cases, but still not in every case.

Reply:
> Maybe something like `llama_n_ctx(&lctx) / llama_n_seq_max(&lctx)`

For Transformer-like models, this would always equal 1.

`llama_n_ctx(&lctx) / cparams.n_seq_max` would be what you meant.